table/block_based/block_based_table_reader_sync_and_async.h

```diff
@@ -274,6 +281,10 @@ DEFINE_SYNC_AND_ASYNC(void, BlockBasedTable::RetrieveMultipleBlocks)
         // through and set up the block explicitly
         if (block_entry->GetValue() != nullptr) {
           s.PermitUncheckedError();
+          if (use_fs_scratch) {
+            // Free the allocated scratch buffer by fs.
+            req.fs_scratch.reset();
```

Comment:
Is it ok to free here? What if the requests have been merged?

Reply:
Oh. If that is the case then I believe I can't release the buffer inside the loop anywhere. It has to be a separate loop outside.